### PR TITLE
Core 171 create is future

### PR DESCRIPTION
**Pull request by tzulberti-jampp**
Based on the logic on Tornado 4.X

Not sure on which is the base branch, because I can not create a PR against a tag

### CODE REVIEW

**Comment by zeitos** [comment]
Is there any kind of testing we can add? is this just a piggyback from tornado 4?

**Comment by tzulberti-jampp** [comment]
The function is tested indirectly on all the code. In tornado 4.X there is no specific test for that function.

File: tornado/gen.py [hunk 1]
```diff
@@ -185,6 +185,45 @@ def coroutine(func):
...
+    function; it is handled by the decorator itself.
+
+    From the caller's perspective, ``@gen.coroutine`` is similar to
+    the combination of ``@return_future`` and ``@gen.engine``.
+
+    This version works with cythonized coroutines by assuming the
+    decorated function is a generator (instead of checking).
+    Use reuglar @gen.coroutine unless necessary.
+    """
+    return _make_coroutine_wrapper(func, assume_generator=True)
```
Comment (klaussfreire) [inline tornado/gen.py:218]: Why is this on the diff?

Something's amiss

Comment (tzulberti-jampp) [reply tornado/gen.py:218]: I choose the branch 3.1 as the base branch. But it seems that the branch doesn't has the commit. I am not sure on which branch it was done

**Comment by tzulberti-jampp** [comment]
Closed this branch and use #2

**Pull request closed without merging**